pkg/wallet/wallet.go

```diff
@@ -697,15 +708,28 @@ func (c *Wallet) Connect(authToken string, invitation *outofband.Invitation, opt
 // 		- DIDCommMsgMap containing request presentation message if operation is successful.
 // 		- error if operation fails.
 //
-func (c *Wallet) ProposePresentation(authToken string, invitation *outofband.Invitation, options ...InitiateInteractionOption) (*service.DIDCommMsgMap, error) { //nolint: lll
+func (c *Wallet) ProposePresentation(authToken string, invitation *GenericInvitation, options ...InitiateInteractionOption) (*service.DIDCommMsgMap, error) { //nolint: lll
 	opts := &initiateInteractionOpts{}
 	for _, opt := range options {
 		opt(opts)
 	}
 
-	connID, err := c.Connect(authToken, invitation, opts.connectOpts...)
-	if err != nil {
-		return nil, fmt.Errorf("failed to perform did connection : %w", err)
+	var connID string
```

Comment:
Looks good, 
but as discussed earlier, may be in some follow up PR, try to put these V1 vs V2 logic inside a wrapper where wallet just initializes right wrapper by passing `GenericInvitation`.